credentials/tls/certprovider/spiffe/spiffe.go

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ * Copyright 2025 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package spiffe implements utilities for grpc-go for work with SPIFFE bundles
+// and SPIFFE bundle maps.
+package spiffe
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/spiffebundle"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+)
+
+type partialParsedSpiffeBundleMap struct {
+	Bundles map[string]json.RawMessage `json:"trust_domains"`
+}
+
+// LoadSpiffeBundleMap loads a SPIFFE Bundle Map from a file. See the SPIFFE
+// Bundle Map spec for more detail -
+// https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Trust_Domain_and_Bundle.md#4-spiffe-bundle-format
+// If duplicate keys are encountered in the JSON parsing, Go's default unmarshal
+// behavior occurs which causes the last processed entry to be the entry in the
+// parsed map.
+//
+// This API is experimental.
```

Comment:
Is this package going to be used only by grpc-go and not for consumption by external users? If so, please move this into an internal directory. Maybe something like `credentials/tls/certprovider/internal/spiffe`.

Also, would it be possible to move the test certs into a subdirectory that lives along with the code, instead of in the top-level testdata directory. Thanks.

Reply:
I can see a case where a user might want to implement their own specific spiffe provider behavior, but that might not actually be reasonable. As for now, this is meant to be used internally.
I found there are actually already a few spiffe utils in internal/credentials/spiffe.go, would you like to see these APIs added there instead?

Reply:
Sure, that makes sense.

And after looking at that file, I think we need to capitalize SPIFFE in the name of this function. See: go/go-style/decisions#initialisms

Reply:
Done